lib/rbvmomi/basic_types.rb

```diff
@@ -164,11 +164,12 @@ def pretty_print q
 class ManagedObject < ObjectWithMethods
   def self.kind; :managed end
 
-  def initialize connection, ref
+  def initialize connection, ref, props={}
     super()
     @connection = connection
     @soap = @connection # XXX deprecated
     @ref = ref
+    @props = Hash[props.map { |k, v| [k.to_sym, v] }]
```

Comment:
This will raise an error if `props` is `nil` (if someone explicitly passes `nil` in). That should probably be handled.